### PR TITLE
OCPBUGS-15215: OAuth template config in HostedCluter.configuration.ouath is not honored in HyperShift

### PR DESCRIPTION
**Pull request by Patryk-Stefanski**
**What this PR does / why we need it**:
Fixes OAuth template config in HostedCluter.configuration.ouath is not honored in HyperShift bug.

**Which issue(s) this PR fixes** :
Fixes #

https://issues.redhat.com/browse/OCPBUGS-15215

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/oauth/deployment.go [hunk 1]
```diff
@@ -97,9 +97,16 @@ func ReconcileDeployment(ctx context.Context, client client.Client, deployment *
 			util.BuildVolume(oauthVolumeKubeconfig(), buildOAuthVolumeKubeconfig),
 			util.BuildVolume(oauthVolumeServingCert(), buildOAuthVolumeServingCert),
 			util.BuildVolume(oauthVolumeSessionSecret(), buildOAuthVolumeSessionSecret),
-			util.BuildVolume(oauthVolumeErrorTemplate(), buildOAuthVolumeErrorTemplate),
-			util.BuildVolume(oauthVolumeLoginTemplate(), buildOAuthVolumeLoginTemplate),
-			util.BuildVolume(oauthVolumeProvidersTemplate(), buildOAuthVolumeProvidersTemplate),
+			util.BuildVolume(oauthVolumeErrorTemplate(), func(volume *corev1.Volume) {
+				buildOAuthVolumeErrorTemplate(volume, params)
+			}),
+			util.BuildVolume(oauthVolumeLoginTemplate(), func(volume *corev1.Volume) {
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/oauth/deployment.go:103]: it'd be nice to have a unit test that validates reconcileDeployment set the right volumes for the right hcp input

Comment (Patryk-Stefanski) [reply control-plane-operator/controllers/hostedcontrolplane/oauth/deployment.go:103]: Do you think this test is sufficient [link](https://github.com/openshift/hypershift/pull/3041/files#diff-34356734f0509551aa83a787c138df16e072fb1938394cbb4c00ba7bbbc12df2R150) ?  We never change the name of the volume being used it's just the secret that changes depending on whether the templates are specified or not and think this should cover it.

**Pull request merged**